### PR TITLE
Add add function to get error count for the current buffer

### PR DESCRIPTION
**Pull request by indelog**
Can be used to display error count in status bar or with vim-airline.

### CODE REVIEW

File: autoload/lsp.vim [hunk 1]
```diff
@@ -502,6 +502,42 @@ def lsp#showCurrentDiag()
   endif
 enddef
 
+# get the count of error in the current buffer
+def lsp#errorCount():dict<number>
+  var res = {'E': 0, 'W': 0, 'I': 0, 'H': 0}
```
Comment (yegappan) [inline autoload/lsp.vim:507]: To make this readable, can you expand this to 'Error', 'Warn', 'Info' and 'Hint'?

File: autoload/lsp.vim [hunk 2]
```diff
@@ -502,6 +502,42 @@ def lsp#showCurrentDiag()
...
+  var ftype = &filetype
+  if ftype == ''
+    return res
+  endif
+
+  var lspserver: dict<any> = s:lspGetServer(ftype)
+  if lspserver->empty()
+    return res
+  endif
+  if !lspserver.running
```
Comment (yegappan) [inline autoload/lsp.vim:517]: Can you combine these two if statements into one?

```
if lspserver->empty() || !lspserver.running
  return res
endif
```

File: autoload/lsp.vim [hunk 3]
```diff
@@ -502,6 +502,42 @@ def lsp#showCurrentDiag()
...
+    return res
+  endif
+
+  var bnr: number = bufnr()
+  if lspserver.diagsMap->has_key(bnr)
+      for item in lspserver.diagsMap[bnr]->values()
+          if item->has_key('severity')
+              if item.severity == 1
+                  res.E = res.E + 1
```
Comment (yegappan) [inline autoload/lsp.vim:526]: This can be simplified to "res.E += 1".

Comment (indelog) [reply autoload/lsp.vim:526]: I can't. If i do this y get an error : `Index with operation not supported yet`

File: autoload/lsp.vim (hunk 1 above)
Comment (indelog) [reply autoload/lsp.vim]: Done

File: autoload/lsp.vim (hunk 2 above)
Comment (indelog) [reply autoload/lsp.vim]: Done

**Pull request merged**